### PR TITLE
Integrate Sprint 6 of Team 1

### PR DESCRIPTION
**Pull request by Erdnaf**
This is the pull request for the work of Team `#1` in Sprint `#6`.

### CODE REVIEW

File: privacy_evaluator/output/user_output_privacy_score.py [hunk 1]
```diff
@@ -1,38 +1,43 @@
...
     samples (outliers) [2]. You could check them again, see if they have the correct label, or if they exhibit any
     non-standard properties for the class. If so, correct them. It was also shown that points with an high influence on
-    the decision boundary are more vulnerable to membership inference attacks [1]. Therefore, these points might be
+    the decision boundary are more vulnerable to membership inference attacks [3]. Therefore, these points might be
     important. If you want to protect them, you might add several similar training samples as they are to the class.
 
     References:
-    [1] Stacey Truex, Ling Liu, Mehmet Emre Gursoy, Lei Yu, and Wenqi Wei. 2019.Demystifying Membership Inference
-    Attacks in Machine Learning as a Service.IEEE Transactions on Services Computing(2019)
+    [1] Song, Liwei and Prateek Mittal. “Systematic Evaluation of Privacy Risks of Machine Learning Models.” ArXiv
```
Comment (fraboeni) [inline privacy_evaluator/output/user_output_privacy_score.py:27]: again, inside a -.py class, I would rather see links to the papers than that reference style.

**Review by fraboeni — CHANGES_REQUESTED** [review]
Hey Team 1, very good job on this sprint. After having addressed my concerns, feel free to merge your changes into the master and clean up the branch! Thank you for all the good work you have done in the project so far.

File: privacy_evaluator/attacks/membership_inference/black_box.py [hunk 2]
```diff
@@ -1,55 +1,62 @@
...
+        y_train: np.ndarray,
+        x_test: np.ndarray,
+        y_test: np.ndarray,
+        **kwargs,
+    ):
         """Fits the attack model.
 
-        :param kwargs: Keyword arguments for the fitting.
+        :param x_train: Data which was used to train the target model and will be used for training the attack model.
+        :param y_train: True, one-hot encoded labels for `x_train`.
```
[the remark this answers is not in the document]
Comment (Friedrich-Mueller) [reply privacy_evaluator/attacks/membership_inference/black_box.py:54]: Thank you for finding that again. You addressed that in the past already and we took care of here (see here https://github.com/privML/privacy-evaluator/pull/189/commits/b5e2e9798e977036f0719785cabc9528d8f537c2). I'm not sure how those got back in tbh.

**Comment by Friedrich-Mueller** [comment]
I removed the 'True's again.
I am working on the references in .py-files and will correct them ASAP. Same goes for the "lowest possible accuracy" which I will update ASAP as well.

This leaves only the feedback regarding the Attacker's Advantage Score description/equation in README.

**Comment by Friedrich-Mueller** [comment]
All of Franziska's comments are addressed now, but still need some peer review.

Especially the one regarding the advantage score equation.

File: README.md [hunk 3]
```diff
@@ -145,18 +143,47 @@ This is the implementation of the original idea of a black-box membership infere
...
+The calculation of the Privacy Risk Score is based on Bayesian Principles and needs both a _(X,y)<sub>target</sub>_ and _(X,y)<sub>non-target</sub>_ dataset in order to work properly.  _(X,y)<sub>non-target</sub>_ is a dataset that was not used to train the target model. For the details of the calculation see the paper linked above.
+
+| Model | Purpose and Dataset |
+|:-------------:|:-------------:|
+| target model | this model was trained on an dataset (X,y)<sub>target</sub> and is used to calculate the privacy risk scores |
+| attack model | no attack model is used for this attack, as a privacy risk score is calculated based on the target model's behaviour with respect to both dataset _(X,y)<sub>target</sub>_ and _(X,y)<sub>non-target</sub>_ |
+
+#### Membership Attacker's Advantage Score
+
+The metric represents the attacker's advantage score.  It represents the degree to which a learning algorithm, i.e. a model, reveals membership to an adversary. The score calculation is based on Definition 4 from [Yeom, Samuel, et al.](https://arxiv.org/pdf/1709.01604.pdf).
```
[the remark this answers is not in the document]
Comment (Friedrich-Mueller) [reply README.md]: I added a simple equation. I'm not sure if its fine like that or if it needs further denotions or explanations.

**Comment by marisanest** [comment]
Thanks @Friedrich-Mueller! I checked everything and it all looks quite good to me. Just added some really small additional explanations to the equation.

**Pull request merged**